diff --git a/libamdf/src/gpu/umd/kfd/target/pm4_queue.c b/libamdf/src/gpu/umd/kfd/target/pm4_queue.c
index 99e57c8dcd..268968adb3 100644
--- a/libamdf/src/gpu/umd/kfd/target/pm4_queue.c
+++ b/libamdf/src/gpu/umd/kfd/target/pm4_queue.c
@@ -10,7 +10,7 @@
 
 enum {
   AMDF_GPU_KFD_PM4_PAGE_SIZE = 4096,
-  AMDF_GPU_KFD_PM4_RING_BYTE_LENGTH = 4096,
+  AMDF_GPU_KFD_PM4_RING_BYTE_LENGTH = 1024 * 1024,
   AMDF_GPU_KFD_PM4_DOORBELL_MAPPING_BYTE_LENGTH = 8192,
 };
 
